nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Cli/Commands/DiscoverCommand.cs

```diff
@@ -26,8 +26,22 @@ internal static Command GetCommand(Action<int> setExitCode)
 
         command.SetHandler(async (jobPath, repoRoot, workspace, outputPath) =>
         {
+            var (experimentsManager, errorResult) = await ExperimentsManager.FromJobFileAsync(jobPath.FullName);
+            if (errorResult is not null)
+            {
+                // to make testing easier, this should be a `WorkspaceDiscoveryResult` object
+                var discoveryErrorResult = new WorkspaceDiscoveryResult
+                {
+                    Path = workspace,
+                    Projects = [],
+                    ErrorType = errorResult.ErrorType,
+                    ErrorDetails = errorResult.ErrorDetails,
+                };
+                await DiscoveryWorker.WriteResultsAsync(repoRoot.FullName, outputPath.FullName, discoveryErrorResult);
+                return;
```

Comment:
We want this to always report error code 0 so that we get a chance to properly elevate the error type and details in `native_helpers.rb` below.